flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/datastream/DataStreamOverAggregate.scala

```diff
@@ -106,6 +112,10 @@ class DataStreamOverAggregate(
         if (overWindow.lowerBound.isUnbounded &&
           overWindow.upperBound.isCurrentRow) {
           createUnboundedAndCurrentRowProcessingTimeOverWindow(inputDS)
+        } // lowerbound is a RowType and upperbound is PRECEEDING or CURRENT ROW
+        else if (overWindow.isRows &&
+            (overWindow.upperBound.isPreceding || overWindow.upperBound.isCurrentRow)) {
```

Comment:
I missed the "not unbounded" clause. However, this check works also for the queries: " BETWEEN 2 ROWS PRECEDING" - letting implicit the current row. Don't you think that being a little flexible on syntax could be useful? If you want to be strict on this, I will apply the check logic as you suggested.